app/main.py

```diff
@@ -1,2 +1,67 @@
+from typing import Any, Hashable, List, Optional
+from fractions import Fraction
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = Fraction(2, 3)
+        self.table: List[Optional[List[Node]]] = [None] * self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+        for bucket in old_table:
+            if bucket:
+                for node in bucket:
```

Comment:
if you do this implementation correctly you will not need to have nested for loops